velox/exec/Task.cpp

```diff
@@ -2590,7 +2590,15 @@ uint64_t Task::MemoryReclaimer::reclaim(
   if (task->isCancelled()) {
     return 0;
   }
-  return memory::MemoryReclaimer::reclaim(pool, targetBytes, maxWaitMs, stats);
+  // Before reclaiming from its operators, first to check if there is any free
+  // capacity in the root after stopping this task.
```

Comment:
Is this needed because memory could have been freed while we waited for the task to pause?

Reply:
Yes, and we didn't check the free memory after that.